### PR TITLE
Migrate test suite from Karma to Jest  🎉🤡

### PR DESCRIPTION
**Pull request by superkat64**
### Description

In the process of writing a test for the new API endpoint, `featureFlagSynchronous`, we noticed the tests and the dependencies were rather outdated. This PR focuses on just migrating the tests from karma to jest and upgrading the javascript since this application hasn't been touched in 7 years 😱

### Changes

- Removed karma, chai, sinon, phantomjs, and mocha from the dependencies
- Added jest to the dev dependencies and setup it's configuration
- Updated `var` to `const` in all test files
- Removed unused imports
- Updated `setTimeout(() => {})` to `async/await`
- Migrated all the old chai and sinon assertions to jest assertions

###### Removed
`chai`
`karma`
`phantomjs`
`mocha`

###### Added
`jest`
`eslint-plugin-jest`

###### Updated
bumped `blueimp-md5` to 2.19.0
bumped  cookies-js` to 1.2.3

### Ticket

_Part of_ [PEG-25](https://customink.atlassian.net/browse/PEG-25)

### Notes

Part of larger version update. This PR is merging into what hopefully with be the `0.5.0` release branch so the stuff to focus on right now is just the tests.

_Recommended reading: [Code Review guide](https://github.com/customink/guides/blob/master/operations/code-review/README.md)_

### Optional Tasks

- [ ] Include 🎩 Instructions
- [ ] Update the readme (README.md)
- [ ] Update the API or architecture docs (e.g. docs/api.md)

### CODE REVIEW

File: .node-version [hunk 1]
```diff
@@ -0,0 +1 @@
+16.4.2
```
Comment (avinash-vllbh) [inline .node-version]: nit: missing new line

File: package.json [hunk 2]
```diff
@@ -29,16 +29,16 @@
...
-    "mocha": "^9.0.0",
+    "eslint-plugin-jest": "^26.0.0",
+    "jest": "^27.4.7",
     "sinon": "^12.0.1",
     "webpack": "^5.0"
   },
   "dependencies": {
-    "blueimp-md5": "1.1.1",
-    "cookies-js": "1.2.2",
+    "blueimp-md5": "2.19.0",
```
Comment (avinash-vllbh) [inline package.json:38]: Were there any changes of note from `1.x` --> `2.x` upgrade?
I took a look at the repo https://github.com/blueimp/JavaScript-MD5 but couldn't find any release/change notes.

**Review by avinash-vllbh — COMMENTED** [review]
LGTM. Not required but I was thinking having Prettier integration will be nice to have to make the formatting consistent across repos.

File: package.json (hunk 2 above)
Comment (superkat64) [reply package.json:38]: I don't believe so, nothing appears to be impacted and it seems like they just started tracking versioning with 2.0.0

**Review by avinash-vllbh — APPROVED** [review]

**Pull request merged**